src/libraries/System.Text.Json/src/System/Text/Json/Serialization/WriteStack.cs

```diff
@@ -88,6 +109,16 @@ internal struct WriteStack
         /// </summary>
         public string? NewReferenceId;
 
+        /// <summary>
+        /// Indicates that the next converter is polymorphic and must serialize a type discriminator.
+        /// </summary>
+        public string? PolymorphicTypeDiscriminator;
```

Comment:
should this be on the frame?

Reply:
It could be -- but at the same time I want to avoid bloating the (already large) stackframe structs with more reference types. Both that and the reference identifier are written at the start of serialization so they don't need to be put on the stack.